### PR TITLE
Plugin template generator

### PR DESCRIPTION
**Pull request by frsyuki**
* Adds `new` subcommand to CLI that generates plugin template.
* Change plugin package naming convention from `embulk-plugin-NAME` to `embulk-CATEGORY-NAME` (e.g. embulk-output-s3)
* Change plugin load path from `embulk/CATEGORY_NAME.rb` to `embulk/CATEGORY/NAME.rb` (e.g. embulk/output/s3.rb)

Usage of plugin generator:

```
$ embulk new
Usage: new <category> <name>
categories:
    ruby-input                 record input plugin    (like mysql)
    ruby-output                record output plugin   (like mysql)
    ruby-filter                record filter plugin   (like add-hostname)
    java-input                 record input plugin    (like mysql)
    java-output                record output plugin   (like mysql)
    java-filter                record filter plugin   (like add-hostname)
    java-file-input            file input plugin      (like ftp)
    java-file-output           file output plugin     (like ftp)
    java-parser                file parser plugin     (like csv)
    java-formatter             file formatter plugin  (like csv)
    java-encoder               file encoder plugin    (like gzip)
    java-decoder               file decoder plugin    (like gzip)

examples:
    new ruby-output hbase
    new ruby-filter int-to-string
```

Example:

```
$ embulk new java-file-input hdfs
Creating embulk-input-hdfs/
  Creating embulk-input-hdfs/README.md
  Creating embulk-input-hdfs/LICENSE.txt
  Creating embulk-input-hdfs/.gitignore
  Creating embulk-input-hdfs/gradle/wrapper/gradle-wrapper.jar
  Creating embulk-input-hdfs/gradle/wrapper/gradle-wrapper.properties
  Creating embulk-input-hdfs/gradlew.bat
  Creating embulk-input-hdfs/gradlew
  Creating embulk-input-hdfs/build.gradle
  Creating embulk-input-hdfs/lib/embulk/input/hdfs.rb
  Creating embulk-input-hdfs/src/main/java/org/embulk/input/HdfsFileInputPlugin.java
  Creating embulk-input-hdfs/src/test/java/org/embulk/input/TestHdfsFileInputPlugin.java

```

### CODE REVIEW

**Comment by frsyuki** [comment]
@nahi  Can you review this pull-request?

File: lib/embulk/command/embulk_new_plugin.rb [hunk 1]
```diff
@@ -0,0 +1,87 @@
+module Embulk
+  def self.new_plugin(name, language, category)
+    require 'embulk/data/package_data'
+    require 'embulk/version'
+    require 'fileutils'
+
+    embulk_category = category
+    embulk_category = :input if category == :file_input
+    embulk_category = :output if category == :file_output
```
Comment (nahi) [inline lib/embulk/command/embulk_new_plugin.rb]: We might want to add file_input and file_output plugins for Ruby in the future? NotImplementedError?

**Comment by nahi** [comment]
Sorry to keep you waiting. Please merge this.

**Comment by muga** [comment]
@frsyuki @nahi this is great!

**Comment by frsyuki** [comment]
@nahi I updated help message.

**Comment by frsyuki** [comment]
Merged.

**Pull request merged**